### PR TITLE
CLOUDP-58634: Add support for checkpoints in the Atlas client

### PR DESCRIPTION
**Pull request by andreaangiolillo**

### CODE REVIEW

File: mongodbatlas/atlas_checkpoints.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"net/http"
+)
+
+const (
+	backupCheckpoints = "groups/%s/clusters/%s/backupCheckpoints"
+)
+
+// AtlasCheckpointService is an interface for interfacing with the Checkpoint
+// endpoints of the MongoDB Atlas API.
+type AtlasCheckpointService interface {
```
Comment (gssbzn) [inline mongodbatlas/atlas_checkpoints.go]: Can we remove the `Atlas` part? This is already the atlas client so it should be clear enoght already

File: mongodbatlas/continuous_snaphots.go [hunk 2]
```diff
@@ -46,16 +46,19 @@ type ContinuousSnapshot struct {
...
+	EncryptionEnabled  bool              `json:"encryptionEnabled,omitempty"`
+	FileSizeBytes      int64             `json:"fileSizeBytes,omitempty"`
+	MasterKeyUUID      string            `json:"masterKeyUUID,omitempty"`
+	MongodVersion      string            `json:"mongodVersion,omitempty"`
+	ReplicaSetName     string            `json:"replicaSetName,omitempty"`
+	StorageSizeBytes   int64             `json:"storageSizeBytes,omitempty"`
+	TypeName           string            `json:"typeName"`
+	ShardName          string            `json:"shardName,omitempty"`
+	TokenDiscovered    bool              `json:"tokenDiscovered,omitempty"`
+	TokenTimestamp     SnapshotTimestamp `json:"tokenTimestamp,omitempty"`
```
Comment (gssbzn) [inline mongodbatlas/continuous_snaphots.go]: Is you want to make this ommitempty you may need to make it a pointer

Comment (andreaangiolillo) [reply mongodbatlas/continuous_snaphots.go]: Do you prefer making them pointers or creating a new struct? I am thinking about the latter

Comment (gssbzn) [reply mongodbatlas/continuous_snaphots.go]: if this is the same for both but with extra fields let's make them pointers

Comment (gssbzn) [reply mongodbatlas/continuous_snaphots.go]: another 3rd way is creating 3 structs, one base, one with only the extra and then one that compose both, composition sometimes is great ;)

File: mongodbatlas/mongodbatlas.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ type Client struct {
 	X509AuthDBUsers                  X509AuthDBUsersService
 	ContinuousSnapshots              ContinuousSnapshotsService
 	ContinuousRestoreJobs            ContinuousRestoreJobsService
-	AtlasCheckpoints                 AtlasCheckpointService
+	AtlasCheckpoints                 CheckpointService
```
Comment (gssbzn) [inline mongodbatlas/mongodbatlas.go]: missed the rename here, can we also make CheckpointService -> CheckpointsService?

File: mongodbatlas/checkpoints.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+//CloudProviderSnapshotRestoreJobsServiceOp handles communication with the CloudProviderSnapshotRestoreJobs related methods of the
+//MongoDB Atlas API
+type CheckpointsServiceOp struct {
+	Client RequestDoer
+}
+
+var _ CheckpointService = &CheckpointsServiceOp{}
+
+type Checkpoint struct {
```
Comment (PacoDw) [inline mongodbatlas/checkpoints.go]: Thank you so much, it's really nice, but could you add comments describing all the importing functions and structs to have coherency? Also, it avoids warnings in some editors for missing comments.

**Pull request merged**